src/palladio/SOPAssign.cpp

```diff
@@ -193,36 +104,103 @@ void updateAttributeUIDefaultValues(SOPAssign* node, const std::wstring& style,
 
 			switch (currParmType.getBasicType()) {
 				case PRM_Type::PRM_BasicType::PRM_BASIC_ORDINAL: {
-					// only support booleans, i.e. don't store folders
-					if (currParmType.getOrdinalType() != PRM_Type::PRM_OrdinalType::PRM_ORD_TOGGLE)
-						continue;
-
-					const int intValue = static_cast<int>(std::get<bool>(it->second));
-
-					node->setInt(attributeName, 0, time, intValue);
-					parm.overwriteDefaults(time);
-					parm.getTemplatePtr()->setFactoryDefaults(parm.getTemplatePtr()->getDefault(0));
+					switch (currParmType.getOrdinalType()) {
```

Comment:
GCC warning: not all enum items handled in switch... consider adding a default case

Reply:
done